### PR TITLE
handwired macropad 4x4 illera16

### PR DESCRIPTION
**Pull request by villeramassana**
## Types of Changes

- [ x] Keyboard addition 

## Checklist

- [ x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ x] I have added tests to cover my changes.
- [ x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/illera16/rules.mk [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# This file intentionally left blank
+
+RGBLIGHT_ENABLE = yes
```
Comment (zvecr) [inline keyboards/handwired/illera16/rules.mk:3]:
```suggestion
```

File: keyboards/handwired/illera16/readme.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# handwired/illera16
+
+![handwired/illera16](imgur.com image replace me!)
```
Comment (zvecr) [inline keyboards/handwired/illera16/readme.md:3]: fill in with valid image or remove

File: keyboards/handwired/illera16/keymaps/vial/config.h
Comment (zvecr) [inline keyboards/handwired/illera16/keymaps/vial/config.h:1]: `vial` keymaps are not accepted to this repo and **must** be removed.

File: keyboards/handwired/illera16/info.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    "maintainer": "villeramassana",
+    "bootloader": "caterina",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (zvecr) [inline keyboards/handwired/illera16/info.json:13]:
```suggestion
        "nkro": true,
        "rgblight": true
```

File: keyboards/handwired/illera16/info.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    },
+    "matrix_pins": {
+        "cols": ["D4", "C6", "D7", "E6"],
+        "rows": ["F4", "F5", "F6", "F7"]
+    },
+    "processor": "atmega32u4",
```
Comment (zvecr) [inline keyboards/handwired/illera16/info.json:19]:
```suggestion
    "development_board": "promicro",
```

File: keyboards/handwired/illera16/info.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{
+    "manufacturer": "victor_illera",
+    "keyboard_name": "illera16",
+    "maintainer": "villeramassana",
+    "bootloader": "caterina",
```
Comment (zvecr) [inline keyboards/handwired/illera16/info.json:5]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/handwired/illera16/info.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        "cols": ["D4", "C6", "D7", "E6"],
+        "rows": ["F4", "F5", "F6", "F7"]
+    },
+    "processor": "atmega32u4",
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0000",
+        "vid": "0xFEED"
+    },
```
Comment (dunk2k) [inline keyboards/handwired/illera16/info.json:25]:
```suggestion
    },
    "community_layouts": ["ortho_4x4"],
```

File: keyboards/handwired/illera16/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+#define _BASE 0 
+#define _CAPA1 1
+#define _CAPA2 2
+#define _CAPA3 3
+#define _CAPA4 4
+#define _CAPA5 5
```
Comment (dunk2k) [inline keyboards/handwired/illera16/keymaps/default/keymap.c:11]:
```suggestion
enum layers {
    _BASE ,
    _CAPA1,
    _CAPA2,
    _CAPA3,
    _CAPA4,
    _CAPA5,
};
```

File: keyboards/handwired/illera16/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+   * | INS |PGUP | c4 |PRSCR|
+   * |-----+-----+-----+-----|
+   * | DEL |PGDN |VOLD |VOLU |
+   * |-----+-----+-----+-----|
+   * |HOME | UP  | END |     |
+   * |-----+-----+-----+-----|
+   * |LEFT |DOWN |RIGHT|ENTER|
+   * `---------------------- '
+   */
+  [5] = LAYOUT_ortho_4x4( /* NAV CLUSTER */
```
Comment (dunk2k) [inline keyboards/handwired/illera16/keymaps/default/keymap.c:116]:
```suggestion
    [_BASE] = LAYOUT_ortho_4x4(
        KC_DEL, TG(1),  ____,   KC_SLASH,
        KC_7,   KC_8,   KC_9,   KC_PAST,
        KC_4,   KC_5,   KC_6,   KC_PMNS,
        KC_1,   KC_2, KC_3, KC_0
    ),
    [_CAPA1] = LAYOUT_ortho_4x4(
        KC_1,     TG(2),   TG(1),   KC_1,
        KC_1,     KC_1,   KC_1,   KC_1,
        KC_1,     KC_1,   KC_1,   KC_1,
        KC_1,     KC_1,   KC_1,   KC_1
    ),
    [_CAPA2] = LAYOUT_ortho_4x4(
        KC_2,      TG(3),   TG(2),   KC_2,
        KC_2,       KC_2,   KC_2,   KC_2,
        KC_2,       KC_2,   KC_2,   KC_2,
        KC_2,       KC_2,   KC_2,   KC_2
    ),
    [_CAPA3] = LAYOUT_ortho_4x4(
        KC_3,      TG(4),   TG(3),   KC_3,
        KC_3,       KC_3,   KC_3,   KC_3,
        KC_3,       KC_3,   KC_3,   KC_3,
        KC_3,       KC_3,   KC_3,   KC_3
    ),
    /* MOUSE
     * ,-------------------------------------------------.
     * |           |           |     c3     |     X      |
     * |-----------+-----------+------------+------------|
     * |  BUTTON 5 |LEFT CLICK |     UP     |RIGHT CLICK |
     * |-----------+-----------+------------+------------|
     * |  BUTTON 4 |   LEFT    |    DOWN    |   RIGHT    |
     * |-----------+-----------+------------+------=-----|
     * |  BUTTON 3 |SCROLL LEFT|SCROLL DOWN |SCROLL RIGHT|
     * `-------------------------------------------------'
     */
    [_CAPA4] = LAYOUT_ortho_4x4(
        _______,   TG(5), TG(4), _______,
        KC_BTN5, KC_BTN1, KC_MS_U, KC_BTN2,
        KC_BTN4, KC_MS_L, KC_MS_D, KC_MS_R,
        KC_BTN3, KC_WH_L, KC_WH_D, KC_WH_R
    ),
     /* NAV
   * ,-----------------------.
   * | INS |PGUP | c4 |PRSCR|
   * |-----+-----+-----+-----|
   * | DEL |PGDN |VOLD |VOLU |
   * |-----+-----+-----+-----|
   * |HOME | UP  | END |     |
   * |-----+-----+-----+-----|
   * |LEFT |DOWN |RIGHT|ENTER|
   * `---------------------- '
   */
  [_CAPA5] = LAYOUT_ortho_4x4(
```

File: keyboards/handwired/illera16/info.json [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+                { "matrix": [2, 2], "x": 2, "y": 2 },
+                { "matrix": [2, 3], "x": 3, "y": 2 },
+                { "matrix": [3, 0], "x": 0, "y": 3 },
+                { "matrix": [3, 1], "x": 1, "y": 3 },
+                { "matrix": [3, 2], "x": 2, "y": 3 },
+                { "matrix": [3, 3], "x": 3, "y": 3 }
+            ]
+        }
+    }
+}
```
Comment (dunk2k) [inline keyboards/handwired/illera16/info.json:48]: Add EoF
```suggestion
}

```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Pull request closed without merging**